datafusion/physical-expr/src/equivalence.rs

```diff
@@ -575,10 +588,19 @@ impl OrderingEquivalenceClass {
         }
     }
 
-    /// Gets the first ordering entry in this ordering equivalence class.
-    /// This is one of the many valid orderings (if there are multiple).
+    /// Gets the concatenated version of the all orderings.
+    /// if orderings are [a ASC, b ASC] and [c ASC, d ASC]
+    /// Returns [a ASC, b ASC, c ASC, d ASC].
+    /// This ensures that during during merging
+    /// [a ASC, b ASC] and [c ASC, d ASC] are still valid.
     pub fn output_ordering(&self) -> Option<LexOrdering> {
-        self.orderings.first().cloned()
+        let output_ordering = self
+            .orderings
+            .iter()
+            .flat_map(|ordering| ordering.to_vec())
+            .collect::<Vec<_>>();
```

Comment:
Nit: We can probably write this more idiomatically and avoid collecting individual pieces.